fuzz/fuzz_targets/msg_targets/utils.rs

```diff
@@ -36,3 +51,84 @@ macro_rules! test_msg_hole {
 		}
 	}
 }
+
+#[macro_export]
+macro_rules! test_msg_writeable {
+	($MsgType: path, $data: ident) => {
+		{
+			use lightning::ln::msgs::{MsgEncodable, MsgDecodable};
+			use lightning::util::ser::{Writer, Reader, Writeable, Readable};
+			let mut r = Reader::new(::std::io::Cursor::new($data));
+			if let Ok(msg) = <$MsgType as Readable<::std::io::Cursor<&[u8]>>>::read(&mut r) {
+				let p = r.get_ref().position() as usize;
+				let mut w = Writer::new(::std::io::Cursor::new(vec![]));
+				msg.write(&mut w).unwrap();
+
+				let buf = w.into_inner().into_inner();
```

Comment:
Hmm, how do most other serialization frameworks work? This looks kinda awkward but maybe this is the normal way to do it? Either way, these functions will need to be non-fuzztarget-only, right?

Reply:
It seems they tend to use `into_inner()` and `get_ref()` when accessing the underlying `Read` like in`std::io::BufReader`. They may be not required in the normal situation, but I'm not totally sure. I agree it shouldn't be non-fuzztarget-only.